core/data/resolve.go

```diff
@@ -110,38 +111,59 @@ type ResolutionDetails struct {
 func GetResolutionDetails(toResolve string) (*ResolutionDetails, error) {
 
 	//todo optimize, maybe tokenize first
+	details := &ResolutionDetails{}
 
-	dotIdx := strings.Index(toResolve, ".")
+	if field.HasSpecialFields(toResolve) {
```

Comment:
What are "Special Fields"

Reply:
This case we use to support special characters in the field name,  user wants to use `.` `$` `^` or any others in field or attr name,  we come up a way to use  [""] or [''] around the field name.  Take example: User wants to get a value from `x.y` node of `attr` field,  the final ref pass to runtime like: $acitivity[acitivityname].attr["x.y"].

Reply:
so attr is a map?

Reply:
It seems like that code should be here, it should be part of path resolution..
The job of GetResolutionDetails is to break it down into four parts...
The resolver (i.e. flow, activity, property)
The item... in case of activity, the activity name
The property attrName, 
and the path  [i]=index, a.b for object traversal, and we should add [""] for maps

Reply:
Thanks @fm-tibco,  I agree to remove this logic to use current way after discussing with you, Vijay and Matt .  the attr name has a special character is not a common way,  we don’t think a  developer would do that